buf.gen.yaml

```diff
@@ -0,0 +1,9 @@
+version: v1
```

Comment:
can all this live inside `tools/src/buf` ?

Reply:
I tried to make this change, but there seems to be a lack of functionality in `buf` to support this. While `buf generate` supports setting a different location for the `buf.yaml` and `buf.gen.yaml` using the `--template` and `--config` flags, it seems to always look for the `buf.lock` file in the root directory. `buf` also expects the `buf.yaml` files to be located alongside the `buf.lock` file so I guess those flags are for use-cases where the `buf` command is not being invoked from the root dir but that you need to point `buf` at where the root dir is. Based on the `buf` docs, does seem like this is known and intended design, but if you figure out a way to make this work then lmk.

Reply:
lets keep buf.lock in the root (and add a comment highlighting the limitation), and move others into src  ?

Reply:
yeah lets stick to 2. for now than, thanks for investing cycles to find a cleaner solution.
would be good to follow up with this in `buf` upstream